packages/web/vite.config.ts

```diff
@@ -79,6 +81,18 @@ export default defineConfig(({ mode }) => {
           }
         }
       },
+      {
+        transformIndexHtml(html) {
+          // Replace HTML env vars with values from the system env
+          Object.keys(APP_ENV).forEach((key) => {
```

Comment:
It might be faster/better to replace all of the matching patterns instead of iterating through each env var. Cause then we can log a warning if any env vars are undefined that we are expecting:

```ts
    return html
      .replace(/%(\S+?)%/g, (text: string, key) => {
        if (key in APP_ENV) {
          return APP_ENV[key as keyof typeof APP_ENV] ?? text
        }
        console.warn(`Missing environment variable: ${key}`)
        return text
      })
 ```

Reply:
It feels a bit less safe to do a match on all strings inside `%`, but the warning is nice. I'll change to it